q2cli/handlers.py

```diff
@@ -310,12 +310,25 @@ def get_click_options(self):
 
     def get_value(self, arguments, fallback=None):
         import qiime2
-
+        import qiime2.sdk.util
         path = self._locate_value(arguments, fallback)
+
         if path is None:
             return None
+
+        try:
+            # check to see if path is an artifact
+            art = qiime2.Artifact.load(path)
+        except Exception:
+            pass
         else:
-            return qiime2.Metadata.load(path)
+            if qiime2.sdk.util.has_metadata(art):
+                return art.view(qiime2.Metadata)
```

Comment:
This should be `qiime2.Metadata.from_artifact(art)` to capture the provenance. (This is the one we undo'ed a bunch, looks like it may not have been redo'ed)